contracts/plugins/trading/DutchTrade.sol

```diff
@@ -183,12 +192,16 @@ contract DutchTrade is ITrade {
         bestPrice = _bestPrice; // gas-saver
     }
 
-    /// Bid for the auction lot at the current price; settling atomically via a callback
-    /// @dev Caller must have provided approval
+    /// Bid with callback for the auction lot at the current price;
+    ///  Sold funds are sent back to the callee, callee.dutchTradeCallback(...) is invoked
+    ///  balance of buy token must increase by bidAmount(current block) after callback
+    ///  Trade is settled atomically via a callback
+    ///
+    /// @param data {bytes} The data to pass to the callback
+    /// @dev Caller must implement IDutchTradeCallee
     /// @return amountIn {qBuyTok} The quantity of tokens the bidder paid
-    function bid() external returns (uint256 amountIn) {
+    function bid(bytes calldata data) external returns (uint256 amountIn) {
```

Comment:
we should probably add nonreentrant to this?

Reply:
I thought of this as well, but I think this function is already non-reentrant because of the `bidder == address(0)` check at the very beginning.